python_bindings/src/PyError.cpp

```diff
@@ -17,6 +17,7 @@ void halide_python_print(JITUserContext *, const char *msg) {
 class HalidePythonCompileTimeErrorReporter : public CompileTimeErrorReporter {
 public:
     void warning(const char *msg) override {
+        py::gil_scoped_acquire acquire;
         py::print(msg, py::arg("end") = "");
```

Comment:
This is now exactly `halide_python_print(nullptr, msg);`. Maybe you want to use that implementation?

Reply:
fixed